### PR TITLE
Extend API with memkind_malloc_usable_size

### PR DESCRIPTION
**Pull request by michalbiesek**
Ref: #58

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/63)

### CODE REVIEW

File: include/memkind.h [hunk 1]
```diff
@@ -282,6 +282,10 @@ int memkind_check_available(memkind_t kind);
 /* malloc from the numa nodes of the specified kind */
 void *memkind_malloc(memkind_t kind, size_t size);
 
+/* EXPERIMENTAL API */
+/* malloc_usable_size from the numa nodes of the specified kind */
```
Comment (pbalcer) [inline include/memkind.h]: "from the numa nodes" ? ;P

Comment (michalbiesek) [reply include/memkind.h]: :) Done

**Comment by kasiawasiuta** [comment]
retest this please

**Comment by michalbiesek** [comment]
___
*[man/memkind.3, line 351 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkbLKK3bP_PeiH-TyV:-LIo5o297sw9xDNDSUAW:byyqub6) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/man/memkind.3#L351)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Unify spaces between function name and parenthesis
</blockquote></details>

Done. For rest of manual files I will create separate PR.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 492 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkbgp--LrY4xIV_Cyq:-LIo6wjk8Zi_Q6kR8wKD:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/src/memkind.c#L492)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

To many new lines
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 497 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkfUDj4MRbeI15KXf6:-LIo6y74EMxj_M-hCYaA:b3t26yc) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/src/memkind.c#L497)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

> ```
> pthread_once
> ```
I'm not sure, but I think that locks in getter function can slow down application to much, if it's not interfere with metadata should be removed
</blockquote></details>

Done - removed

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 174 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkcVGz8aZTxuqK2or0:-LIo703H4wQEMmcQ9tf_:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/test/memkind_pmem_tests.cpp#L174)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Check_sizes is a instantiation of anonymous structure, so their name should start from small letter
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 187 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkcuIgEa2D2XtWdrN-:-LIo71qh2D2Yl68PF8Ea:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/test/memkind_pmem_tests.cpp#L187)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

no space before assignment sign
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 194 at r2](https://reviewable.io/reviews/memkind/memkind/63#-LIkcRMx6V76q9j42kVS:-LIo72s32OtaWxNBltop:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/578178ce562943464286dcf3e6a7639c83b917e8/test/memkind_pmem_tests.cpp#L194)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

No space after assignment
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 194 at r3](https://reviewable.io/reviews/memkind/memkind/63#-LIoe4AnCzMou6XOhWlg:-LIoikRHAWDD6YNCiQ3-:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7cb132b10543d6a1cbf056ba27d763b490cb5562/test/memkind_pmem_tests.cpp#L194)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

++i :)
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 497 at r3](https://reviewable.io/reviews/memkind/memkind/63#-LIoceR7BkI4a4X2fp6h:-LIoimMn9xLz3Bt7ZZ3Z:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7cb132b10543d6a1cbf056ba27d763b490cb5562/src/memkind.c#L497)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

add whitespace between arguments
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[man/memkind.3, line 352 at r3](https://reviewable.io/reviews/memkind/memkind/63#-LIobeG_3-cRfMo5Oh39:-LIoinyJCHxgiJgtqpM1:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7cb132b10543d6a1cbf056ba27d763b490cb5562/man/memkind.3#L352)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

I would write "returns". Above there is also "return", but I think we should fix that too
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[include/memkind/internal/memkind_private.h, line 97 at r3](https://reviewable.io/reviews/memkind/memkind/63#-LIoafdSEpFaqIe5bk27:-LIoipODDeGAMaDLLERU:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7cb132b10543d6a1cbf056ba27d763b490cb5562/include/memkind/internal/memkind_private.h#L97)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

> ```
>     void (*init_once)(void);
>     int (* finalize)(struct memkind *kind);
>     size_t (*malloc_usable_size)(struct memkind *kind, void *addr);
> ```
Pls unify whitespaces
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[include/memkind.h, line 286 at r1](https://reviewable.io/reviews/memkind/memkind/63#-LIQavaYmEDmOMVankSj-r1-286:-LIogMFIDIGUi3jQ6WpR:b4vz3uu) ([raw file](https://github.com/memkind/memkind/blob/222143ff94d89eee1151fa71589fa87528803a47/include/memkind.h#L286)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

Maybe "of memory block"?
</blockquote></details>

http://man7.org/linux/man-pages/man3/malloc_usable_size.3.html 
"obtain size of block of memory allocated"

**Pull request merged**